### PR TITLE
docs: update Pagination

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

Current Pagination docs: https://ui.docs.amplify.aws/components/pagination

Updated Pagination docs (demo):
pagination

The Pagination docs required an overhaul in order to demonstrate its usage as a controlled component. The first few examples are complete, and I've simplified the code for the styling examples. 

The most effective way to review this PR would be to run it locally and see if anything in the Pagination docs needs to be changed. 

#### Issue #, if available
https://github.com/aws-amplify/amplify-ui/issues/937

#### Description of how you validated changes
Running the docs locally. 

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by reesscot — COMMENTED** [review]
Great work, just had a few suggestions

File: docs/src/pages/components/pagination/react.mdx [hunk 1]
```diff
@@ -60,83 +99,56 @@ To replace the Pagination styling, unset it:
   all: unset;
   /* Add your styling here*/
 }
-```
+````
```
[the remark this answers is not in the document]
Comment (joebuono) [reply docs/src/pages/components/pagination/react.mdx:102]: Yeah it should just be three. Not sure how it got formatted like this

File: docs/src/pages/components/pagination/react.mdx [hunk 2]
```diff
@@ -60,83 +99,56 @@ To replace the Pagination styling, unset it:
...
-```jsx
-import { Pagination } from '@aws-amplify/ui-react';
-import '@aws-amplify/ui-react/styles.css';
-import './styles.css';
+<Example>
+  <PaginationStylingExample className="my-custom-pagination" totalPages={7} />
+  <ExampleCode>
+    ```css
+    /* styles.css */
+    .my-custom-pagination {
```
[the remark this answers is not in the document]
Comment (joebuono) [reply docs/src/pages/components/pagination/react.mdx:115]: We are pulling from the actual CSS file. This is just to display the code in the demo

File: docs/src/pages/components/pagination/react.mdx [hunk 3]
```diff
@@ -68,75 +142,56 @@ To override styling on a specific Pagination component, you can use a class sele
...
+  </ExampleCode>
+</Example>
 
 ## Accessibility
 
-By default, the root node has a role of "navigation" and each page item has an aria-label that identifies the purpose of the item ("Go to previous page", "Go to next page", "Go to page 1" etc.).
-````
+By default, the root node of the Pagination component has a role of "navigation", and each page item has an `aria-label` that identifies the purpose of the item ("Go to previous page", "Go to next page", "Go to page 1", etc).
+
+```
```
Comment (reesscot) [inline docs/src/pages/components/pagination/react.mdx:195]: remove empty code section?

**Review by jacoblogan — APPROVED** [review]
LGTM

File: docs/src/pages/components/pagination/PaginationPropControls.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  totalPages,
+  setTotalPages,
+  siblingCount,
+  setSiblingCount,
+  hasMorePages,
+  setHasMorePages,
+}) => {
+  return (
+    <Flex direction="column">
+      <StepperField
```
Comment (0618) [inline docs/src/pages/components/pagination/PaginationPropControls.tsx:42]: nit: maybe we can map the `StepperField`?

File: docs/src/pages/components/pagination/examples/ControlledPaginationExample.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { Pagination } from '@aws-amplify/ui-react';
+
+export const ControlledPaginationExample = () => {
+  const [currentPageIndex, setCurrentPageIndex] = React.useState(1);
+  const totalPages = 5;
+
+  const handleNextPage = () => {
+    console.log('handleNextPage');
```
Comment (0618) [inline docs/src/pages/components/pagination/examples/ControlledPaginationExample.tsx:9]: Why do we have the `console.log`s in this file?

File: docs/src/pages/components/pagination/examples/PaginationThemeExample.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  usePagination,
+} from '@aws-amplify/ui-react';
+
+const theme = {
+  name: 'pagination-theme',
+  tokens: {
+    components: {
+      pagination: {
+        current: {
+          backgroundColor: { value: 'rebeccapurple' },
```
Comment (0618) [inline docs/src/pages/components/pagination/examples/PaginationThemeExample.tsx:13]: nit: do we need to use the tokens?

File: docs/src/pages/components/pagination/react.mdx [hunk 7]
```diff
@@ -37,7 +59,22 @@ It's common to use a paged API where the total number of pages in the dataset is
...
+</Example>
+
+### Sibling Count
+
+`siblingCount` is an optional prop which controls the number of pages displayed on each side of the current page (defaults to `1`). For example, in the demo below, notice how page 5 has two siblings to the left (3 and 4) and two siblings to the right (6 and 7).
+
+<Example>
+  <PaginationSiblingCountExample />
+  <ExampleCode>
+```jsx{8} file=./examples/PaginationSiblingCountExample.tsx
```
Comment (0618) [inline docs/src/pages/components/pagination/react.mdx:74]: curious: what does `{8}` do?

**Review by 0618 — APPROVED** [review]

File: docs/src/pages/components/pagination/react.mdx (hunk 7 above)
Comment (joebuono) [reply docs/src/pages/components/pagination/react.mdx:74]: Highlights line 8 in the mdx

File: docs/src/pages/components/pagination/examples/ControlledPaginationExample.tsx (hunk 5 above)
Comment (joebuono) [reply docs/src/pages/components/pagination/examples/ControlledPaginationExample.tsx:9]: To help show the developer when these events are triggered

**Pull request merged**